drivers/entropy/entropy_stm32.c

```diff
@@ -357,6 +357,24 @@ static int entropy_stm32_rng_init(const struct device *dev)
 	__ASSERT_NO_MSG(dev_cfg != NULL);
 
 #if CONFIG_SOC_SERIES_STM32L4X
+
+#if !defined(CONFIG_CLOCK_STM32_PLL_SRC_MSI) && \
+	!defined(CONFIG_CLOCK_STM32_PLL_SRC_HSI) && \
+	!defined(CONFIG_CLOCK_STM32_PLL_SRC_HSE)
+
+	/* If no PLLSRC clock, use other available 48 MHz clock source */
+#if defined(CONFIG_CLOCK_STM32_SYSCLK_SRC_HSI)
+	LL_RCC_SetRNGClockSource(LL_RCC_RNG_CLKSOURCE_HSI48);
+#elif defined(CONFIG_CLOCK_STM32_SYSCLK_SRC_MSI) && \
+	defined(CONFIG_CLOCK_STM32_MSI_RANGE) && \
+	CONFIG_CLOCK_STM32_MSI_RANGE == 11
+	LL_RCC_SetRNGClockSource(LL_RCC_RNG_CLKSOURCE_MSI);
```

Comment:
I dig a bit more and I have several remarks:
1) I could not find a restriction in reference manual of STM32L4, to have MSI clock at 48MHZ.
Any MSI value seems valid for RNG.
The only restriction is when USB is also used in parallel, as they share the same clock and USB require 48MHz.
But it is up to USB to manage this restriction. 
2) I am not sure to understand what was wrong with the previous code:
It seems there is a full configuration of PLLSAI1 to feed the RNG.
Maybe just miss to enable MSI (LL_RCC_MSI_Enable) ?
```
	/* Configure PLLSA11 to enable 48M domain */
	LL_RCC_PLLSAI1_ConfigDomain_48M(LL_RCC_PLLSOURCE_MSI,
					LL_RCC_PLLM_DIV_1,
					24, LL_RCC_PLLSAI1Q_DIV_2);

	/* Enable PLLSA1 */
	LL_RCC_PLLSAI1_Enable();

	/*  Enable PLLSAI1 output mapped on 48MHz domain clock */
	LL_RCC_PLLSAI1_EnableDomain_48M();

	/* Wait for PLLSA1 ready flag */
	while (LL_RCC_PLLSAI1_IsReady() != 1) {
	}

	/*  Write the peripherals independent clock configuration register :
	 *  choose PLLSAI1 source as the 48 MHz clock is needed for the RNG
	 *  Linear Feedback Shift Register
	 */
	 LL_RCC_SetRNGClockSource(LL_RCC_RNG_CLKSOURCE_PLLSAI1);
```

Reply:
1. 1. Thanks for the input, the code is even simpler without MSI frequency restrictions.

Reply:
2. In our STM32L462 system we chose to use MSI @ 48 MHz as system clock and we have both PLL and PLLSAI1 disabled.
This means having PLLSRC=00 ("No clock sent to PLL PLLSAI1" and even PLL) that corresponds having no CONFIG_CLOCK_STM32_PLL_SRC_xxx defined in Zephyr.
With the previous code, after programming PLLSAI1, the loop to wait for PLLSA1 ready flag was hanging.
Yesterday I was supposing that the previous code was not working just because we have no PLL programmed (that's why of compile switch "#if !defined(CONFIG_CLOCK_STM32_PLL_SRC_MSI) && ..."). 
After reading your note, I remembered that in the past days I already noticed that apparently the previous code should work, but we didn't find the reason why it's looping.
On the other hand, we don't want to enable PLL and we already have a 48 MHz source, so the natural decision was to use MSI.
Now I believe to have found why the previous code doesn't work: it probably assumes to have MSI @ 4 MHz and it's multiplying it by 24 to have 48 MHz; instead it's multiplying 48 MHz *24!
In conclusion, the current solution uses only the existing configuration to obtain the following logic: if no PLL is wanted, either use MSI or HSI as clock source for RNG, otherwise use PLL.